demos/free_surface/explicit_viscous_surface.py

```diff
@@ -0,0 +1,140 @@
+
+    # Function to store the solutions:
+    z = Function(Z)  # a field over the mixed function space Z
+    u, p = split(z)  # Returns symbolic UFL expression for u and p
+    u_, p_ = z.subfunctions  # Do this first to extract individual velocity and pressure fields
+
+    # Next rename for output:
+    u_.rename("Velocity")
+    p_.rename("Pressure")
+
+    eta = Function(W, name="eta")  # Define a free surface function
+
+    T = Function(Q, name="Temperature").assign(0)  # Setup a dummy function for temperature
+
+    # Output function space information:
+    log("Number of Velocity DOF:", V.dim())
+    log("Number of Pressure DOF:", W.dim())
+    log("Number of Velocity and Pressure DOF:", V.dim()+W.dim())
+    log("Number of Temperature DOF:", Q.dim())
+
+    eta_eq = FreeSurfaceEquation(W, W, free_surface_id=top_id)  # Initialise the separate free surface equation for explicit coupling
+
+    # Stokes related constants (note that since these are included in UFL, they are wrapped inside Constant):
+    Ra = Constant(0)  # Rayleigh number, here we set this to zero as there are no bouyancy terms
+    approximation = BoussinesqApproximation(Ra)
+
+    rho0 = approximation.rho  # This defaults to rho0 = 1 (dimensionless)
+    g = approximation.g  # This defaults to g = 1 (dimensionless)
+
+    kk = Constant(2 * pi / lam)  # wavenumber (dimensionless)
+    F0 = Constant(1000 / L0)  # initial free surface amplitude (dimensionless)
+    X = SpatialCoordinate(mesh)
+    eta.interpolate(F0 * cos(kk * X[0]))  # initial free surface condition
+
+    # timestepping
+    mu = Constant(1)  # Dynamic viscosity (dimensionless)
+    tau0 = Constant(2 * kk * mu / (rho0 * g))  # Characteristic time scale (dimensionless)
+    log("tau0", tau0)
+
+    dt = Constant(0.5*dt_factor*tau0)  # Timestep (dimensionless)
+    log("dt (dimensionless)", dt)
+
+    time = Constant(0.0)
+    max_timesteps = round(10*tau0/dt)  # Simulation runs for 10 characteristic time scales so end state is close to being fully relaxed
+    log("max_timesteps", max_timesteps)
+
+    # No normal flow except on the free surface
+    stokes_bcs = {
+        bottom_id: {'un': 0},
+        top_id: {'free_surface_stress': rho0 * g * eta},  # Apply stress on free surface
+        left_id: {'un': 0},
+        right_id: {'un': 0},
+    }
+
+    eta_fields = {'velocity': u_}
+
+    eta_bcs = {}
+    eta_strong_bcs = [InteriorBC(W, 0., top_id)]  # Apply strong homogenous boundary to interior DOFs to prevent a singular matrix when only integrating the free surface equation over the top surface.
+
+    mumps_solver_parameters = {
+        'snes_monitor': None,
+        'snes_type': 'ksponly',
+        'ksp_type': 'preonly',
+        'pc_type': 'lu',
+        'pc_factor_mat_solver_type': 'mumps',
+        'mat_type': 'aij',
+        'snes_max_it': 100,
+        'snes_rtol': 1e-8,
+        'snes_atol': 1e-6,
+        'mat_mumps_icntl_14': 200
+    }
```

Comment:
How is this different from the defaults in `StokesSolver` ? Why do you need a Newton solver here - isn't this a linear system? If you do need additional options, in addition to the defaults you can do:
```
  stokes_solver = StokesSolver(...)  # without solver_parameters, sets default options
  stokes_solver.solver_parameters.update({
      # only options that are different from the defaults
   })
```

Reply:
thanks I meant to go back to the defaults ( I had copied these from Thwaites at one point when I was debugging)